include/algorithms/synthesis/encoding.hpp

```diff
@@ -38,8 +38,8 @@ namespace syrec {
 
     auto extend(TruthTable& tt) -> void;
 
-    auto encodeHuffman(TruthTable& tt) -> std::pair<TruthTable::CubeMap, std::size_t>;
+    auto encodeHuffman(TruthTable& tt, bool log2MaxOccurrences = false) -> std::pair<TruthTable::CubeMap, std::size_t>;
 
-    auto augmentWithConstants(TruthTable& tt, std::size_t const& nBits, bool dc = false) -> void;
+    auto augmentWithConstants(TruthTable& tt, std::size_t const& nBits, bool dc = false, bool appendZero = false, bool appendZeroIn = false) -> void;
```

Comment:
I am not a big fan of endlessly adding flags to functions. This makes them hard to read and hardly ever better to use. Furthermore, AFAICT not all of these combinations are actually used. Consider refactoring this interface a little.